spec/tree_spec.rb

```diff
@@ -1,14 +1,240 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+describe Tree do
+  let(:tree) { Tree.new }
+
+  it 'should be an AppleTree' do
+    expect(tree.is_a? Tree).to be true
+  end
+
+  it 'should have a height of 0' do
+    expect(tree.height).to be 0  
+  end
+
+  it 'should have an age of 0' do
+    expect(tree.age).to be 0
+  end
+
+  it 'should have an empty set of apples' do
+    expect(tree.apples).to eq []
+  end
+
+  describe '#age!' do
+    
+    context 'after aging once' do
+      before(:each) { tree.age! }
+
+      it 'age is 1' do
+        expect(tree.age).to be 1
+      end
+
+      it 'height is 1' do
+        expect(tree.height).to be 1
+      end
+
+      it 'has 5 apples' do
+        expect(tree.apples.length).to be 5     
+      end
+
+    end
+    
+    context 'after aging twice' do
+      before(:each) { tree.age!; tree.age! }
+
+      it 'age is 2' do
+        expect(tree.age).to be 2
+      end
+      
+      it 'height is 2' do
+        expect(tree.height).to be 2
+      end
+
+      it 'has 15 apples' do
+        expect(tree.apples.length).to eq 15
+      end
```

Comment:
I'm thinking that the tests for aging twice might be redundant with respect to the tests for one aging. My thought process was that more would be better, and the second tests would ensure that the age method wasn't simply setting the height and/or age to 1 (as absurd as that sounds to do in the context of this problem).